kroxylicious-integration-tests/src/test/java/io/kroxylicious/kms/service/KmsIT.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Abstract Kms IT.
+ *
+ * @param <C> The config type
+ * @param <K> The key reference
+ * @param <E> The type of encrypted DEK
+ */
+@SuppressWarnings("java:S5960") // this is test code, it exists in the main module to facilitate its use by concrete test cases
+
+@ExtendWith(RecordEncryptionTestInvocationContextProvider.class)
+class KmsIT<C, K, E> {
+
+    private Kms<K, E> kms;
+
+    private TestKekManager manager;
+
+    KmsIT() {
+    }
+
+    @BeforeEach
+    void beforeEach(TestKmsFacade<?, ?, ?> facade) {
+        facade.start();
+        try {
+            var service = (KmsService<C, K, E>) facade.getKmsServiceClass().getDeclaredConstructor(new Class[]{}).newInstance();
+            kms = service.buildKms((C) facade.getKmsServiceConfig());
+        }
+        catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+        manager = facade.getTestKekManager();
+    }
+
+    @TestTemplate
+    void resolveKeyByName() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .isEqualTo(alias);
+    }
+
+    @TestTemplate
+    void resolveWithUnknownKey() {
+        var alias = "unknown";
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .failsWithin(Duration.ofSeconds(5))
+                .withThrowableThat()
+                .withCauseInstanceOf(UnknownAliasException.class);
+    }
+
+    @TestTemplate
+    void generatedEncryptedDekDecryptsBackToPlain() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+
+        var resolveStage = kms.resolveAlias(alias);
+        assertThat(resolveStage).succeedsWithin(Duration.ofSeconds(5));
```

Comment:
I think this assert obscures the real "assertion" this test case is making and I think we can remove it as we already cover this step in `resolveKeyByName`.